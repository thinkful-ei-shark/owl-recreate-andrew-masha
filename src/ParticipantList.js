import React from 'react'
import Participant from './Participant.js'

function ParticipantList(props) {
  const participants = props.list.map(participant => (
  <Participant key={participant.id} props={participant} />
    )
  )
  return (
    <ul className="Participant-list">
      {participants}
    </ul>
  )
}

export default ParticipantList;