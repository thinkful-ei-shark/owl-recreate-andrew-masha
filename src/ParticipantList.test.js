import React from 'react';
import ParticipantList from './ParticipantList.js';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

const testObj = [
  {
      id: 1,
      name: 'Koren Templeton',
      avatar:
          'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: true
  },
  {
      id: 2,
      name: 'Caty Flucker',
      avatar:
          'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: false
  },
  {
      id: 3,
      name: 'Axe Kubicka',
      avatar: 'https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1',
      inSession: false,
      onStage: false
  },
  {
      id: 4,
      name: 'Frank Runciman',
      avatar:
          'https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: false
  },
  {
      id: 5,
      name: 'Ashla Attwool',
      avatar:
          'https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: true
  }
];

it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(
    <ParticipantList list={testObj} />,
    div
  );

  ReactDOM.unmountComponentAtNode(div);
})


it('renders the UI as expected', () => {
  const tree = renderer
    .create(<ParticipantList list={testObj} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});