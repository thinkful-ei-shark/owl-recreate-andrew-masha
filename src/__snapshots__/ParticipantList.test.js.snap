// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<ul
  className="Participant-list"
>
  <li
    className="Participant"
  >
    <div
      className="portait"
    >
      <img
        alt="avatar"
      />
    </div>
    <div
      className="participant-meta"
    >
      <div
        className="participant-name"
      />
      <div
        className="Participant-byline"
      >
        <span
          className="dot not-in-session"
        />
        <span
          className="stage-status"
        >
          Off Stage
        </span>
      </div>
    </div>
  </li>
  <li
    className="Participant"
  >
    <div
      className="portait"
    >
      <img
        alt="avatar"
      />
    </div>
    <div
      className="participant-meta"
    >
      <div
        className="participant-name"
      />
      <div
        className="Participant-byline"
      >
        <span
          className="dot not-in-session"
        />
        <span
          className="stage-status"
        >
          Off Stage
        </span>
      </div>
    </div>
  </li>
  <li
    className="Participant"
  >
    <div
      className="portait"
    >
      <img
        alt="avatar"
      />
    </div>
    <div
      className="participant-meta"
    >
      <div
        className="participant-name"
      />
      <div
        className="Participant-byline"
      >
        <span
          className="dot not-in-session"
        />
        <span
          className="stage-status"
        >
          Off Stage
        </span>
      </div>
    </div>
  </li>
  <li
    className="Participant"
  >
    <div
      className="portait"
    >
      <img
        alt="avatar"
      />
    </div>
    <div
      className="participant-meta"
    >
      <div
        className="participant-name"
      />
      <div
        className="Participant-byline"
      >
        <span
          className="dot not-in-session"
        />
        <span
          className="stage-status"
        >
          Off Stage
        </span>
      </div>
    </div>
  </li>
  <li
    className="Participant"
  >
    <div
      className="portait"
    >
      <img
        alt="avatar"
      />
    </div>
    <div
      className="participant-meta"
    >
      <div
        className="participant-name"
      />
      <div
        className="Participant-byline"
      >
        <span
          className="dot not-in-session"
        />
        <span
          className="stage-status"
        >
          Off Stage
        </span>
      </div>
    </div>
  </li>
</ul>
`;
